Lib/poplib.py

```diff
@@ -424,6 +427,88 @@ def stls(self, context=None):
         self._tls_established = True
         return resp
 
+    def auth(self, mechanism, authobject, *, initial_response_ok=True):
+        """Authenticate to the POP3 server using AUTH (RFC 5034).
+
+        Result is 'response'.
+        """
+        mech = mechanism.upper()
+
+        initial = None
+        if initial_response_ok:
+            try:
+                initial = authobject()
+            except TypeError:
+                initial = None
```

Comment:
```suggestion
        if initial_response_ok and callable(authobject):
            initial = authobject()
```

I would prefer that we don't silence exceptions raised by user-defined functions. If the user provides a callable that requires more parameters, we should raise normally.